File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    basket = []
    count = 0
    
    for col in moves:
        col -= 1
        row = 0
        while row < len(board)-1 and board[row][col] == 0:
            row += 1
        
        if board[row][col] != 0:
            basket.append(board[row][col])
            board[row][col] = 0
            if len(basket) > 1 and basket[-1] == basket[-2]:
                basket.pop()
                basket.pop()
                count += 2
    
    return count

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.swift
import Foundation

func solution(_ board:[[Int]], _ moves:[Int]) -> Int {
    var board = board
    var basket = [Int]()
    var result = 0
    
    for move in moves {
        var row = 0
        let col = move - 1
        
        while row < board.count {
            if board[row][col] == 0 {
                row += 1
            } else {
                break
            }
        }
        
        guard row < board.count else { continue }
        
        basket.append(board[row][col])
        board[row][col] = 0
        
        let basketMax = basket.count
        if basketMax >= 2 
        && basket[basketMax - 1] == basket[basketMax - 2] {
            basket.removeLast()
            basket.removeLast()
            result += 2
        }
        
    }
    
    return result
}